app/javascript/packages/rails-i18n-webpack-plugin/rails-i18n-webpack-plugin.js

```diff
@@ -111,48 +45,44 @@ class RailsI18nWebpackPlugin extends ExtractKeysWebpackPlugin {
    * Cached locale data.
    *
    * @type {{
-   *   [domain: string]: {
-   *     [locale: string]: Promise<{ [key: string]: string }>
-   *   }
+   *   [locale: string]: Promise<{ [key: string]: string }>
    * }}
    */
   localeData = Object.create(null);
 
   /**
    * Given a translation domain and locale, returns the file path corresponding to locale data.
    *
-   * @param {string} domain
    * @param {string} locale
    *
-   * @return {string}
+   * @return {string[]}
    */
-  getLocaleFilePath(domain, locale) {
-    return path.resolve(this.options.configPath, domain, `${locale}.yml`);
+  getLocaleFilePaths(locale) {
+    return /** @type {string[]} */ (readdirSync(this.options.configPath, { recursive: true }))
+      .filter((/** @type {string} */ filePath) => filePath.endsWith(`${locale}.yml`))
+      .map((filePath) => path.resolve(this.options.configPath, filePath));
   }
 
   /**
    * Returns a promise resolving to parsed YAML data for the given domain and locale.
    *
-   * @param {string} domain
    * @param {string} locale
    *
    * @return {Promise<undefined|Record<string, string>>}
    */
-  getLocaleData(domain, locale) {
-    if (!(domain in this.localeData)) {
-      this.localeData[domain] = Object.create(null);
-    }
-
-    if (!(locale in this.localeData[domain])) {
-      const localePath = this.getLocaleFilePath(domain, locale);
-
-      this.localeData[domain][locale] = fs
-        .readFile(localePath, 'utf-8')
-        .then(YAML.parse)
-        .catch(() => {});
+  getLocaleData(locale) {
+    if (!(locale in this.localeData)) {
+      this.localeData[locale] = Promise.all(
+        this.getLocaleFilePaths(locale).map((filePath) =>
+          fs
+            .readFile(filePath, 'utf-8')
+            .then(YAML.parse)
+            .catch(() => {}),
+        ),
+      ).then((keys) => /** @type {Record<string, string>} */ Object.assign({}, ...keys));
```

Comment:
Not sure I follow what's happening here, we're creating a shallow clone of the object?

Reply:
English has multiple i18n files so this merges all files in a locale to one object.

(I left transliteration and telephony separate)

Reply:
Ohhh, I misread the code, I thought `keys` was an object, didn't realize `keys` was an array of objects that we're merging here. All good then 👍

Reply:
Open to renaming `keys` if you have suggestions!

Reply:
Renamed `keys` to `fileDatas` in f30a45b7c2cd533e1df3acb1eed0f6ce7febb7bf